tests/unit/clients/test_gdns.py

```diff
@@ -86,26 +86,31 @@ def test_get_rrsets_as_objects(rrset_dict):
 
 
 @pytest.mark.asyncio
+@pytest.mark.parametrize('params', [
+    None,
+    {'fields': 'some-other-field,another-field'},
+    {'name': 'example.com.', 'type': 'A'}
+])
 async def test_get_records_for_zone(fake_response_data, client, caplog,
-                                    monkeypatch):
-    mock_get_json_called = 0
+                                    mocker, params):
+    mock_get_json_called = False
 
     async def mock_get_json(*args, **kwargs):
         nonlocal mock_get_json_called
         data = fake_response_data.copy()
         if not mock_get_json_called:
             data['nextPageToken'] = 1
-        mock_get_json_called += 1
+            mock_get_json_called = True
         return data
 
-    monkeypatch.setattr(client, 'get_json', mock_get_json)
+    client.get_json = mock_get_json
 
     url = f'{client._base_url}/managedZones/a-zone/rrsets'
     with aioresponses() as mocked:
         mocked.get(url, status=200)
         # paginated requests
         mocked.get(url, status=200)
-        records = await client.get_records_for_zone('a.zone.')
+        records = await client.get_records_for_zone('a.zone.', params=params)
```

Comment:
Should we be testing that await self.get_json(url, params=params) was called with the right params?